winch/codegen/src/isa/x64/regs.rs

```diff
@@ -81,6 +81,10 @@ pub(crate) fn scratch() -> Reg {
     r11()
 }
 
+pub(crate) fn argv() -> Reg {
```

Comment:
Perhaps we should add a comment here on the use and implications of this register? Something like:

```
This register is used as a scratch register, in the context of trampolines only, where we assume that callee-saved registers are given the correct handling according to the system ABI. r12 is chosen given that it's a callee-saved, non-argument register.

In the context of all other internal functions, this register is not excluded from register allocation, so no extra assumptions should be made regarding its availability.
```

Reply:
This looks good!